### ISSUE TITLE
Istio sidecar keeps pointing to a stale dst host 

### ISSUE THREAD

**Issue by parthshah86** [istio/istio#23267]
**Bug description**
We have istio sidecar running alongside pods within a deployment. The pods communicate with the istio gateway via the sidecar. One pattern which we have observed is that GET request from the pod keeps giving 503 intermittently and does not recover without restarting the pod. 

Looking at the gateway logs, the request does not hit the gateway. In the istio sidecar logs, one thing which we observe is that the sidecars point to an IP that no longer exist. This can happen if pods dst hosts recycle. The sidecar does not seem to recover from this issue. 

**Expected behavior**
In-case the Dst host goes down, istio sidecar should switch to the new host IP address. 

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
`istioctl version --remote`
client version: 1.4.5
control plane version: 1.4.3
data plane version: 1.4.3 (1 proxies)

`kubectl version`
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.4", GitCommit:"8d8aa39598534325ad77120c120a22b3a990b5ea", GitTreeState:"clean", BuildDate:"2020-03-12T21:03:42Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.9-eks-502bfb", GitCommit:"502bfb383169b124d87848f89e17a04b9fc1f6f0", GitTreeState:"clean", BuildDate:"2020-02-07T01:31:02Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}

`helm version`
Client: &version.Version{SemVer:"v2.14.1", GitCommit:"5270352a09c7e8b6e8c9593002a73535276507c0", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}

**Environment where bug was observed (cloud vendor, OS, etc)**
EKS

**Comment by howardjohn** [comment]
Please upgrade to 1.4.8 this bug was likely fixed by https://github.com/istio/istio/issues/20676

**Comment by parthshah86** [comment]
Thanks @howardjohn, I will try upgrading the version.

**Comment by parthshah86** [comment]
The upgrade seems to have fixed this issue. Thanks @howardjohn.

**Comment by parthshah86** [comment]
The issue does not seem to be completely resolved with istio 1.5.2. We started getting the issue again. 
With the same setup where a container is talking to an external service via the istio proxy we are getting 503 continuously. These are the log messages from the istio proxy. 

```
[Envoy (Epoch 0)] [2020-05-15 05:00:35.837][18][trace][connection] [external/envoy/source/common/network/connection_impl.cc:409] [C3] writing 209 bytes, end_stream false
[Envoy (Epoch 0)] [2020-05-15 05:00:35.837][18][trace][http] [external/envoy/source/common/http/conn_manager_impl.cc:1764] [C3][S5951578934124360947] encoding data via codec (size=16 end_stream=true)
[Envoy (Epoch 0)] [2020-05-15 05:00:35.837][18][trace][connection] [external/envoy/source/common/network/connection_impl.cc:409] [C3] writing 27 bytes, end_stream false
[Envoy (Epoch 0)] [2020-05-15 05:00:35.837][18][trace][main] [external/envoy/source/common/event/dispatcher_impl.cc:162] item added to deferred deletion list (size=1)
[Envoy (Epoch 0)] [2020-05-15 05:00:35.837][18][trace][http] [external/envoy/source/common/http/conn_manager_impl.cc:985] [C3][S5951578934124360947] decode headers called: filter=0x55eb2c17d9e0 status=1
[Envoy (Epoch 0)] [2020-05-15 05:00:35.837][18][trace][http] [external/envoy/source/common/http/http1/codec_impl.cc:522] [C3] parsed 130 bytes
[Envoy (Epoch 0)] [2020-05-15 05:00:35.837][18][trace][connection] [external/envoy/source/common/network/connection_impl.cc:473] [C3] socket event: 2
[Envoy (Epoch 0)] [2020-05-15 05:00:35.837][18][trace][connection] [external/envoy/source/common/network/connection_impl.cc:561] [C3] write ready
[Envoy (Epoch 0)] [2020-05-15 05:00:35.837][18][trace][connection] [external/envoy/source/common/network/raw_buffer_socket.cc:67] [C3] write returns: 236
[Envoy (Epoch 0)] [2020-05-15 05:00:35.837][18][trace][main] [external/envoy/source/common/event/dispatcher_impl.cc:79] clearing deferred deletion list (size=1)
[Envoy (Epoch 0)] [2020-05-15 05:00:37.288][18][trace][upstream] [external/envoy/source/common/upstream/original_dst_cluster.cc:146] Stale original dst hosts cleanup triggered.
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][debug][pool] [external/envoy/source/common/http/http1/conn_pool.cc:343] [C19965] connect timeout
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][debug][connection] [external/envoy/source/common/network/connection_impl.cc:101] [C19965] closing data_to_write=0 type=1
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][debug][connection] [external/envoy/source/common/network/connection_impl.cc:192] [C19965] closing socket: 1
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][debug][client] [external/envoy/source/common/http/codec_client.cc:91] [C19965] disconnect. resetting 0 pending requests
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][debug][pool] [external/envoy/source/common/http/http1/conn_pool.cc:136] [C19965] client disconnected, failure reason: 
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][debug][pool] [external/envoy/source/common/http/http1/conn_pool.cc:166] [C19965] purge pending, failure reason: 
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][debug][router] [external/envoy/source/common/router/router.cc:990] [C20][S14186131194710689961] upstream reset: reset reason connection failure
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][debug][http] [external/envoy/source/common/http/conn_manager_impl.cc:1418] [C20][S14186131194710689961] Sending local reply with details upstream_reset_before_response_started{connection failure}
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][trace][http] [external/envoy/source/common/http/conn_manager_impl.cc:1505] [C20][S14186131194710689961] encode headers called: filter=0x55eb2b594a00 status=0
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][trace][http] [external/envoy/source/common/http/conn_manager_impl.cc:1505] [C20][S14186131194710689961] encode headers called: filter=0x55eb2b405c70 status=0
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][debug][filter] [src/envoy/http/mixer/filter.cc:135] Called Mixer::Filter : encodeHeaders 2
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][trace][http] [external/envoy/source/common/http/conn_manager_impl.cc:1505] [C20][S14186131194710689961] encode headers called: filter=0x55eb2ba09e00 status=0
[Envoy (Epoch 0)] [2020-05-15 05:00:37.496][25][debug][http] [external/envoy/source/common/http/conn_manager_impl.cc:1616] [C20][S14186131194710689961] encoding headers via codec (end_stream=false):
':status', '503'
'content-length', '91'
'content-type', 'text/plain'
'date', 'Fri, 15 May 2020 05:00:37 GMT'
'server', 'envoy'
```

**Comment by romilpunetha** [comment]
@parthshah86 How did you fix this? We're facing the same issue.
